libs/design-system-components/src/components/AppCreationForm/index.tsx

```diff
@@ -0,0 +1,189 @@
+import Stack from '@akashaorg/design-system-core/lib/components/Stack';
+import Divider from '@akashaorg/design-system-core/lib/components/Divider';
+import DropDown from '@akashaorg/design-system-core/lib/components/Dropdown';
+import { apply, tw } from '@twind/core';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { AkashaAppApplicationType } from '@akashaorg/typings/lib/sdk/graphql-types-new';
+import { ButtonType } from '../types/common.types';
+
+export enum FieldName {
+  extensionType = 'extensionType',
+  extensionID = 'extensionID',
+  extensionName = 'extensionName',
+}
+
+type AppCreationFormValues = {
+  extensionType: { id: string; type: AkashaAppApplicationType; title: string };
+  extensionID: '';
+  extensionName: '';
+};
+
+export type AppCreationFormProps = {
+  defaultValues?: AppCreationFormValues;
+  cancelButton: ButtonType;
+  createButton: {
+    label: string;
+    loading?: boolean;
+    handleClick: (data: AppCreationFormValues) => void;
+  };
+  errorMessage?: { fieldName: string; message: string };
+};
+
+const AppCreationForm: React.FC<AppCreationFormProps> = ({
+  defaultValues = {
+    extensionType: { id: '1', type: AkashaAppApplicationType.App, title: 'Application' },
+    extensionID: '',
+    extensionName: '',
+  },
+  cancelButton,
+  createButton,
+  errorMessage,
+}) => {
+  const {
+    control,
+    setError,
+    clearErrors,
+    getValues,
+    formState: { errors, dirtyFields },
+  } = useForm<AppCreationFormValues>({
+    defaultValues,
+    resolver: zodResolver(schema),
+    mode: 'onChange',
+  });
+  React.useEffect(() => {
+    if (!!errorMessage) {
+      setError(errorMessage.fieldName as FieldName, {
+        type: 'custom',
+        message: errorMessage.message,
+      });
+    }
+  }, [errorMessage, errors]);
+
+  const extensionTypes = [
+    { id: '1', type: AkashaAppApplicationType.App, title: 'Application' },
+    { id: '2', type: AkashaAppApplicationType.Other, title: 'Extension' },
+    { id: '3', type: AkashaAppApplicationType.Widget, title: 'Widget' },
+  ];
+  const isFormDirty =
+    Object.keys(dirtyFields).includes(FieldName.extensionID) &&
+    Object.keys(dirtyFields).includes(FieldName.extensionName);
+  const isValid = !Object.keys(errors).length;
+
+  const onSave = (event: SyntheticEvent) => {
+    event.preventDefault();
+    const formValues = getValues();
+    if (isValid && isFormDirty) {
+      createButton.handleClick({
+        ...formValues,
+      });
+    }
+  };
+
+  return (
+    <form onSubmit={onSave} className={tw(apply`h-full`)}>
+      <Stack direction="column" spacing="gap-y-4">
+        <Stack padding="px-4 pb-16" spacing="gap-y-4">
+          <Controller
+            control={control}
+            name={FieldName.extensionType}
+            render={({ field: { name, value = extensionTypes[0], onChange, ref } }) => (
```

Comment:
You have default value set, setting value to `extensionTypes[0]` shouldn't be necessary.

Reply:
thanks for noticing it Didd.